src/sqlfluff/dialects/dialect_clickhouse.py

```diff
@@ -1146,6 +1146,562 @@ class RenameStatementSegment(BaseSegment):
     )
 
 
+class DictionarySourceSegment(BaseSegment):
+    """A `SOURCE` clause for a ClickHouse dictionary."""
+
+    type = "dictionary_source"
+    match_grammar = Sequence(
+        "SOURCE",
+        Bracketed(
+            OneOf(
+                # CLICKHOUSE
+                Sequence(
+                    "CLICKHOUSE",
+                    Bracketed(
+                        AnyNumberOf(
+                            OneOf(
+                                # Standard parameters
+                                Sequence(
+                                    Ref("NakedIdentifierSegment"),
+                                    OneOf(
+                                        Ref("QuotedLiteralSegment"),
+                                        Ref("NumericLiteralSegment"),
+                                        Ref("BooleanLiteralGrammar"),
+                                    ),
+                                ),
+                                # UPDATE_FIELD parameter
+                                Sequence(
+                                    OneOf(
+                                        "UPDATE_FIELD",
+                                        "update_field",
```

Comment:
This lower case isn't needed here, we define all keywords in the syntax in upper case